tools/ci_connector_ops/ci_connector_ops/pipelines/utils.py

```diff
@@ -73,57 +110,78 @@ async def get_file_contents(container: Container, path: str) -> Optional[str]:
         return await container.file(path).contents()
     except QueryError as e:
         if "no such file or directory" not in str(e):
-            # this is the hicky bit of the stopgap because
             # this error could come from a network issue
             raise
     return None
 
 
-# This is a stop-gap solution to capture non 0 exit code on Containers
-# The original issue is tracked here https://github.com/dagger/dagger/issues/3192
+async def get_container_output(container: Container) -> Tuple[str, str]:
+    """Retrieve both stdout and stderr of a container, concurrently.
+
+    Args:
+        container (Container): The container to execute.
+
+    Returns:
+        Tuple[str, str]: The stdout and stderr of the container, respectively.
+
+    Raises:
+        ExecError: If the container exit code is not 0.
+    """
+    async with asyncer.create_task_group() as task_group:
+        soon_stdout = task_group.soonify(container.stdout)()
```

Comment:
`ExecError` in running `container.stdout` or `container.stderr` will raise `ExceptionGroup` and not `ExecError`. 
The upstream code calling `get_container_output` is handling `ExecError` not `ExceptionGroup`.

The original `with_exit_code` , `with_stdout` etc. Captured the QueryError themselves and returned their string representation, so no `ExceptionGroup` were ever thrown in case of a `QueryError`.